src/main/kotlin/uk/gov/justice/digital/hmpps/pecs/jpc/config/SecurityConfiguration.kt

```diff
@@ -64,19 +72,25 @@ class SecurityConfiguration : WebSecurityConfigurerAdapter() {
       val user = delegate.loadUser(userRequest)
       val jwt = JwtDecoders.fromIssuerLocation(issuer).decode(userRequest.accessToken.tokenValue)
 
-      DefaultOAuth2User(jwt.getClaimAsStringList("authorities").stream().map { SimpleGrantedAuthority(it) }.toList(), user.attributes, "name")
+      DefaultOAuth2User(
+        jwt.getClaimAsStringList("authorities").stream().map { SimpleGrantedAuthority(it) }.toList(),
+        user.attributes,
+        "name"
+      )
     }
   }
 
   private fun accessDeniedHandler(): AccessDeniedHandler {
-    return AccessDeniedHandler { request, _, accessDeniedException ->
+    return AccessDeniedHandler { request, response, accessDeniedException ->
       val auth: Authentication? = SecurityContextHolder.getContext().authentication
 
-      if (auth != null) {
-        logger.warn("User: ${auth.name} attempted to access the protected URL: ${request.requestURI}")
-      }
+      auth?.let { logger.warn("User: ${auth.name} attempted to access the protected URL: ${request.requestURI}") }
 
       logger.error(accessDeniedException.message)
+
+      request.session.invalidate()
```

Comment:
Without this an unused session will sit around waiting for it to he cleaned up.  There is no need as this can be cleaned up straight away.